### PR TITLE
Feature/validation

### PR DESCRIPTION
**Pull request by lagartoverde**

### CODE REVIEW

**Comment by lagartoverde** [comment]
The service output is 100% on dutch, so I guess the validation errors should also be in dutch? In this case could someone translate them?.
If they should be in both english and dutch I think we should send the language in the request and do some logic in the service.

File: support/behandeling-exporter.js [hunk 1]
```diff
@@ -52,7 +58,18 @@ function wrapZittingInfo(zitting, behandelingHTML) {
     .readFileSync(path.join(__dirname, "templates/behandeling-prepublish.hbs"))
     .toString();
   const template = Handlebars.compile(templateStr);
-  return template({behandelingHTML, zitting, prefixes: prefixes.join(" ")});
+  const html = template({behandelingHTML, zitting, prefixes: prefixes.join(" ")});
+  const errors = []
+  if(!zitting.geplandeStart) {
+    errors.push('You must set the planned start of the meeting')
```
Comment (nvdk) [inline support/behandeling-exporter.js:64]:
```suggestion
    errors.push('You must set the planned start of the meeting');
```
missing semi

File: support/behandeling-exporter.js [hunk 2]
```diff
@@ -52,7 +58,18 @@ function wrapZittingInfo(zitting, behandelingHTML) {
...
+  }
+  if(!zitting.start) {
+    errors.push('You must set the start of the meeting')
```
Comment (nvdk) [inline support/behandeling-exporter.js:67]: missing semi

File: support/behandeling-exporter.js [hunk 3]
```diff
@@ -52,7 +58,18 @@ function wrapZittingInfo(zitting, behandelingHTML) {
...
+  }
+  if(!zitting.end) {
+    errors.push('You must set the end of the meeting')
```
Comment (nvdk) [inline support/behandeling-exporter.js:70]: missing semi

File: support/behandeling-exporter.js [hunk 4]
```diff
@@ -52,7 +58,18 @@ function wrapZittingInfo(zitting, behandelingHTML) {
...
+  }
+  return {html, errors}
```
Comment (nvdk) [inline support/behandeling-exporter.js:72]: missing semi

File: support/behandeling-exporter.js [hunk 5]
```diff
@@ -71,6 +88,29 @@ function generateBehandelingHTML(agendapunt) {
...
+    .toString();
+  const template = Handlebars.compile(templateStr);
+  const behandelingUri = agendapunt.behandeling.uri;
+  const agendapuntUri = agendapunt.uri;
+  const agendapuntTitle = agendapunt.title;
+  const openbaar = agendapunt.behandeling.openbaar === 'true' ? true : false;
+  const document = agendapunt.behandeling.document.content;
+  const documentNode = new jsdom.JSDOM(document).window.document;
+  const documentContainer = documentNode.querySelector(`[property='prov:generated']`)
+  const isBesluit = documentContainer.getAttribute('typeof').includes('besluit:Besluit')
```
Comment (nvdk) [inline support/behandeling-exporter.js:103]: what if no container was found? e.g what if this doesn't have a decision

File: support/besluit-exporter.js [hunk 6]
```diff
@@ -56,9 +56,12 @@ async function wrapZittingInfo(besluitenlijst, zitting) {
     .readFileSync(path.join(__dirname, "templates/besluitenlijst-prepublish.hbs"))
     .toString();
   const template = Handlebars.compile(templateStr);
-  const output=template({besluitenlijst, zitting, prefixes: prefixes.join(" ")});
-  
-  return output;
+  const html = template({besluitenlijst, zitting, prefixes: prefixes.join(" ")});
+  const errors = []
+  if(!zitting.geplandeStart) {
+    errors.push('You must set the planned start of the meeting')
```
Comment (nvdk) [inline support/besluit-exporter.js:62]: missing semi

File: support/besluit-exporter.js [hunk 7]
```diff
@@ -56,9 +56,12 @@ async function wrapZittingInfo(besluitenlijst, zitting) {
     .readFileSync(path.join(__dirname, "templates/besluitenlijst-prepublish.hbs"))
     .toString();
   const template = Handlebars.compile(templateStr);
-  const output=template({besluitenlijst, zitting, prefixes: prefixes.join(" ")});
-  
-  return output;
+  const html = template({besluitenlijst, zitting, prefixes: prefixes.join(" ")});
+  const errors = []
```
Comment (nvdk) [inline support/besluit-exporter.js:60]: missing semi

File: support/notule-exporter.js [hunk 8]
```diff
@@ -22,7 +22,18 @@ async function extractNotulenContentFromZitting(zitting, publicBehandelingUris)
   let behandelingsHtml = '';
   behandelingsHtml = generateBehandelingHtml(zitting, publicBehandelingUris);
   const notulenData = Object.assign(zitting, {behandelingsHtml, prefixes: prefixes.join(' ')});
-  return generateNotulenHtml(notulenData);
+  const html = generateNotulenHtml(notulenData);
+  const errors = [];
+  if(!zitting.geplandeStart.value) {
+    errors.push('You must set the planned start of the meeting')
```
Comment (nvdk) [inline support/notule-exporter.js:28]: missing semi

File: support/notule-exporter.js [hunk 9]
```diff
@@ -22,7 +22,18 @@ async function extractNotulenContentFromZitting(zitting, publicBehandelingUris)
...
+  }
+  if(!zitting.startedAt.value) {
+    errors.push('You must set the start of the meeting')
```
Comment (nvdk) [inline support/notule-exporter.js:31]: missing semi

File: support/notule-exporter.js [hunk 10]
```diff
@@ -22,7 +22,18 @@ async function extractNotulenContentFromZitting(zitting, publicBehandelingUris)
...
+  }
+  if(!zitting.endedAt.value) {
+    errors.push('You must set the end of the meeting')
```
Comment (nvdk) [inline support/notule-exporter.js:34]: missing semi

File: support/notule-exporter.js [hunk 11]
```diff
@@ -22,7 +22,18 @@ async function extractNotulenContentFromZitting(zitting, publicBehandelingUris)
   let behandelingsHtml = '';
   behandelingsHtml = generateBehandelingHtml(zitting, publicBehandelingUris);
   const notulenData = Object.assign(zitting, {behandelingsHtml, prefixes: prefixes.join(' ')});
-  return generateNotulenHtml(notulenData);
+  const html = generateNotulenHtml(notulenData);
+  const errors = [];
+  if(!zitting.geplandeStart.value) {
```
Comment (nvdk) [inline support/notule-exporter.js:27]: this check is quite similar from the one in wrapZittingInfo, might make sense to abstract it and reuse in both places

File: support/notule-exporter.js [hunk 12]
```diff
@@ -37,7 +48,9 @@ function generateBehandelingHtml(zitting, publicBehandelingUris) {
   let behandelingHTML = '';
   for(const agendapunt of zitting.agendapunten) {
     if(!publicBehandelingUris || publicBehandelingUris.includes(agendapunt.behandeling.uri)) {
-      behandelingHTML += createBehandelingExtract(zitting, agendapunt, false);
+      behandelingHTML += createBehandelingExtract(zitting, agendapunt, false, true);
+    } else {
+      behandelingHTML += createBehandelingExtract(zitting, agendapunt, false, false)
```
Comment (nvdk) [inline support/notule-exporter.js:53]: missing semi

**Pull request merged**